docs/composition.ipynb

```diff
@@ -0,0 +1,739 @@
+    "    output_measure=dp.max_divergence(T=float),\n",
+    "    #d_in is the upper bound on distances\n",
+    "    d_in=1,\n",
+    "    #d_mids is the privacy consumption for each query\n",
+    "    d_mids=[1., 1., 2.]\n",
+    ")\n",
+    "\n",
+    "#this is the privacy consumption of the entire composition\n",
+    "print(\"Privacy Consumption of Entire Composition: \" + str(sc_meas.map(1)))\n",
+    "\n",
+    "#clean and compile data\n",
+    "hours_list = list(np.array(df.select(\n",
+    "                                    pl.col(\"HWUSUAL\").\n",
+    "                                    fill_null(40.0).\n",
+    "                                    fill_nan(40.0).\n",
+    "                                    filter(pl.col(\"ILOSTAT\")==1)).\n",
+    "                                    collect()\n",
+    "                                ).reshape(-1))\n",
+    "\n",
+    "#this is a queryable that takes in the data similar to the context\n",
+    "sc_qbl = sc_meas(hours_list)\n",
+    "#we can then apply the queryable to a function of our choosing\n",
+    "sc_qbl "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "DP count:  79853\n",
+      "DP Mean: 40.20028441533665\n"
+     ]
+    }
+   ],
+   "source": [
+    "#construct measurement for mean\n",
+    "#compute dp median as an input for dp mean \n",
+    "input_space = dp.vector_domain(dp.atom_domain(T=float)), dp.symmetric_distance()\n",
+    "count_meas = input_space >> dp.t.then_count() >> dp.m.then_laplace(scale=1.0)\n",
+    "count_release = sc_qbl(count_meas)\n",
+    "print(\"DP count: \",count_release)\n",
+    "\n",
+    "mean_meas = (\n",
+    "    input_space >> \n",
+    "    dp.t.then_clamp((0.,79.)) >> \n",
+    "    dp.t.then_resize(size=count_release, constant=40.) >> \n",
+    "    dp.t.then_mean() >> \n",
+    "    dp.m.then_laplace(1.))\n",
+    "print(\"DP Mean:\", sc_qbl(mean_meas))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "DP Median: 37\n"
+     ]
+    }
+   ],
+   "source": [
+    "#construct measurement for median\n",
+    "discrete_scores = dp.t.make_quantile_score_candidates(dp.vector_domain(dp.atom_domain(T=float)), \n",
+    "                                    dp.symmetric_distance(), \n",
+    "                                    np.array(candidates).astype(float), \n",
+    "                                    0.5)\n",
+    "input_space = dp.vector_domain(dp.atom_domain(T=dp.usize), 31), dp.linf_distance(T=dp.usize)\n",
+    "select_index_measurement = dp.m.make_report_noisy_max_gumbel(*input_space, scale=1.0, optimize='min')\n",
+    "\n",
+    "median_meas = discrete_scores >> select_index_measurement >> (lambda index: candidates[index])\n",
```

Comment:
Getting down to the noisy_max_gumbel distribution, etc. At other points, this is an implementation detail the user doesn't need to see. It's not clear to me here if you're using it to demonstrate how things work at a lower level... or if this is something that actually would be the best solution for some usecase.

Reply:
it's a part of the measurement to create a DP median, so across the different types of compositors dp mean and dp median are reported, just through different methods.